rir/src/ir/CodeStream.h

```diff
@@ -21,34 +21,32 @@ class CodeStream {
 
     unsigned pos = 0;
     unsigned size = 1024;
+    unsigned nops = 0;
 
     FunctionWriter& function;
 
     SEXP ast;
 
     unsigned nextLabel = 0;
-    std::map<unsigned, BC::Label> patchpoints;
-    std::vector<unsigned> label2pos;
 
-    std::vector<unsigned> sources;
+    // Ordered map on purpose, see FunctionWriter
+    std::map<unsigned, BC::Label> patchpoints;
+    std::map<unsigned, std::list<BC::Label>> labels;
```

Comment:
I'd prefer `std::vector`. Conventional wisdom is that it's a good enough default and `std::list` is rarely the better choice. If we really think there's a difference and `std::list` is better, we should profile. Until then, I don't think it's worth changing the "default."